pandas/tseries/resample.py

```diff
@@ -323,6 +323,11 @@ def aggregate(self, arg, *args, **kwargs):
                                                *args,
                                                **kwargs)
 
+        # if arg was a string, _aggregate called resampler's _downsample or
+        # _groupby_and_agg methods, which would've already applied the loffset
+        if not isinstance(arg, compat.string_types):
```

Comment:
And as for the implementation, for something as small as `loffset`, is it really worth refactoring a lot of code to change the way it is applied?

I've gone through the stack trying to find another place to apply it, but this is by far the clearest & simplest way I've found.